app/src/main/java/com/hieuwu/groceriesstore/presentation/authentication/signin/SignInViewModel.kt

```diff
@@ -40,6 +40,10 @@ class SignInViewModel @Inject constructor(
         _password.value = newPassword
     }
 
+    fun isValidEmail(): Boolean {
+        return android.util.Patterns.EMAIL_ADDRESS.matcher(_email.value).matches()
```

Comment:
Please import thí one for simplicity

```suggestion
import android.util.Patterns.EMAIL_ADDRESS
        return EMAIL_ADDRESS.matcher(_email.value).matches()
```